src/statistics.jl

```diff
@@ -327,7 +327,7 @@ function apply_statistic(stat::HistogramStatistic,
             x_min = Gadfly.concrete_minimum(values)
             span = x_max - x_min
             binwidth = span / d
-            bincounts ./= sum(bincounts) * binwidth
+            bincounts = bincounts ./ sum(bincounts) * binwidth
```

Comment:
Note that your new 0.4-style code is still type-unstable.  If I were you, I'd use a different variable name (`binmean`?) for the normalized `bincounts`.